include/circt/Dialect/OM/Evaluator/Evaluator.h

```diff
@@ -337,6 +337,27 @@ struct PathValue : EvaluatorValue {
   StringAttr field;
 };
 
+/// An unknown value, i.e., a value that we don't care about evaluating.  During
+/// evaluation, this acts like a "poison" value which will cause any operation
+/// it is folded with to become an invalid value.  The use for this is to
+/// "partially evaluate" a object where you don't care about values for all
+/// input ports.
+struct UnknownValue : EvaluatorValue {
+  /// Create a value representing unknown information.  This value is _always_
+  /// fully evaluated and finalized.  There is no further processing required.
+  UnknownValue(MLIRContext *ctx)
+      : EvaluatorValue(ctx, Kind::Unknown, UnknownLoc::get(ctx)) {
```

Comment:
Might it make sense to support optionally providing a location?  It's stored anyway, so might as well remember where the unknown-ness value "came from", maybe?